lib/livebook/unique_task.ex

```diff
@@ -74,7 +74,7 @@ defmodule Livebook.UniqueTask do
     receive do
       {:DOWN, ^ref, :process, ^pid, reason} ->
         case reason do
-          :normal -> :ok
+          :shutdown -> :ok
```

Comment:
```suggestion
          :normal -> :ok
          :shutdown -> :ok
          {:shutdown, _} -> :ok
```

In case our tasks call other things that exit in the future. :)

Reply:
Would that sill be semantically successful run? The idea is that we call a function, it finished successfully and then we `:shutdown`, and if we get another shutdown then the function hasn't finished, no?

Reply:
I would consider all of the above to be successful/clean shutdowns.

The other option is to have the task reply to this process at the end with `{ref, :ok}`. Then if you get the reply, everything after is ok. Otherwise everything else is an error.

Reply:
If a task for the given key is already running we monitor it instead, so there can be multiple processes monitoring, that's why we rely on exit reason.

Reply:
So my suggestion is the way to go and it aligns with the OTP conventions. :)